parastell/invessel_build.py

```diff
@@ -536,6 +517,221 @@ def generate_rib(self):
+            del self.radial_build['sol']
+        elif self._wall_s > 1.0 and 'sol' not in self._radial_build:
+            self._radial_build = {
+                'sol': {
+                    'thickness_matrix': np.zeros((
+                        len(self._toroidal_angles),
+                        len(self._poloidal_angles)
+                    ))
+                },
+                **self._radial_build
+            }
+        self._radial_build = {
+            'plasma': {
+                'thickness_matrix': np.zeros((
+                    len(self._toroidal_angles),
+                    len(self._poloidal_angles)
+                ))
+            },
+            **self._radial_build
+        }
+        
+        for name, component in self._radial_build.items():
+            if (
+                component['thickness_matrix'].shape !=
+                (len(self._toroidal_angles), len(self._poloidal_angles))
+            ):
+                e = AssertionError(
+                    f'The dimensions of {name}\'s thickness matrix '
+                    f'{component["thickness_matrix"].shape} must match the '
+                    'dimensions defined by the toroidal and poloidal angle '
+                    'lists '
+                    f'{len(self._toroidal_angles),len(self._poloidal_angles)}, '
+                    'which define the rows and columns of the matrix, '
+                    'respectively.'
+                )
+                self._logger.error(e.args[0])
+                raise e
+            
+            if np.any(np.array(component['thickness_matrix']) < 0):
+                e = ValueError(
+                    'Component thicknesses must be greater than or equal to 0. '
+                    'Check thickness inputs for negative values.'
+                )
+                self._logger.error(e.args[0])
+                raise e
+
+            if 'mat_tag' not in component:
+                self._set_mat_tag(name, name)
+
+    @property
+    def logger(self):
+        return self._logger
+    
+    @logger.setter
+    def logger(self, logger_object):
+        self._logger = logger_object
+        if self._logger == None or not self._logger.hasHandlers():
+            self._logger = log.init()
+
+    @property
+    def plasma_mat_tag(self):
+        return self._plasma_mat_tag
+    
+    @plasma_mat_tag.setter
+    def plasma_mat_tag(self, mat_tag):
+        self._plasma_mat_tag = mat_tag
+        self._set_mat_tag('plasma', self._plasma_mat_tag)
+
+    @property
+    def sol_mat_tag(self):
+        return self._sol_mat_tag
+    
+    @sol_mat_tag.setter
+    def sol_mat_tag(self, mat_tag):
+        self._sol_mat_tag = mat_tag
+        self._set_mat_tag('sol', self._sol_mat_tag)
+
+    def _set_mat_tag(self, name, mat_tag):
+        """Sets material tag for a given component.
+        (Internal function not intended to be called externally)
+        """
+        self._radial_build[name]['mat_tag'] = mat_tag
```

Comment:
Should this check to see if the component `name` has a `thickness` matrix already?  Maybe it doesn't matter?  It only matters that the `thickness` matrix is there once you aim to use it...

Reply:
The setter for `radial_build` checks that each component has a `'thickness_matrix'` of the proper shape, which indirectly checks that a `'thickness_matrix'` exists.